code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForCalendarAuthorizationServlet.java

```diff
@@ -0,0 +1,87 @@
+import com.google.api.services.gmail.model.ListLabelsResponse;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.auth.oauth2.AuthorizationRequestUrl;
+import java.util.HashMap;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import com.fasterxml.jackson.core.type.TypeReference;
+import javax.servlet.RequestDispatcher;
+import com.google.api.services.calendar.CalendarScopes;
+import java.io.PrintWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+import java.util.List;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import java.util.Optional;
+
+@WebServlet("/check-for-calendar-credentials")
+public class CheckForCalendarAuthorizationServlet extends HttpServlet {
+
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String TOKENS_DIRECTORY_PATH = "tokens";
+  private static final List<String> SCOPES = Collections.singletonList(CalendarScopes.CALENDAR_READONLY);
+  private static final String CREDENTIALS_FILE_PATH = "/credentials.json";
+  
+  
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+  
+      InputStream in = CheckForCalendarAuthorizationServlet.class.getResourceAsStream(CREDENTIALS_FILE_PATH);
+          if (in == null) {
+              throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+          }
+      GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+      //if the request does not have these request properties, indicating it has not created a credential, it will be directed to create credentials
+      //otherwise it'll access those credentials
+      if (request.getAttribute("authUrlRequestProperties") == null) {
+        Optional<Payload> payload = CheckForCredentials.getPayload(request, response);
+        CheckForCredentials.createCredentials(request, response, SCOPES, "check-for-contacts-credentials", payload);
+      } else {
+        AuthorizationRequestUrl authUrlRequestProperties = (AuthorizationRequestUrl) request.getAttribute("authUrlRequestProperties");
+        GoogleAuthorizationCodeFlow flow = (GoogleAuthorizationCodeFlow) request.getAttribute("flow");
+
+        ObjectMapper mapper = new ObjectMapper();
+        Map<String, Object> authUrlProperties = mapper.readValue(authUrlRequestProperties.getState(), new TypeReference<Map<String, Object>>() {});
+        String userId = (String) authUrlProperties.get("userId");
+        Credential credential = flow.loadCredential(userId);
+        if(credential == null) {
+          PrintWriter out = response.getWriter();
+          out.println("Something went wrong, please proceed to manual input.");
```

Comment:
instead, can you use the Response sendError rather than priting it? Give it a 500 or a 403 Error code.